src/Microsoft.AspNetCore.Blazor/Components/BindInputElementAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
+    public sealed class BindInputElementAttribute : Attribute
+    {
+        public BindInputElementAttribute(string type, string suffix, string valueAttribute, string changeHandlerAttribute)
+        {
+            if (valueAttribute == null)
+            {
+                throw new ArgumentNullException(nameof(valueAttribute));
+            }
+
+            if (changeHandlerAttribute == null)
+            {
+                throw new ArgumentNullException(nameof(changeHandlerAttribute));
+            }
+
+            Type = type;
+            Suffix = suffix;
+            ValueAttribute = valueAttribute;
+            ChangeHandlerAttribute = changeHandlerAttribute;
+        }
+        
+        public string Type { get; }
+        
+        public string Suffix { get; }
+
+        public string ValueAttribute { get; }
+
+        public string ChangeHandlerAttribute { get; }
```

Comment:
This is just a property name. For the code that handles the case of `<input />` this will have the value `onchange`. I chose the naming because I wanted to make a distintion between **the value** and **the change handler** as concepts. The naming isn't intended to match the actual attributes.

Reply:
OK

Reply:
I'm going to just simplify this to `ValueAttribute` + `ChangeAttribute`. I think it's clearer without extra nouns.